regression/cbmc/array-function-parameters/test.c

```diff
@@ -0,0 +1,20 @@
+#include <assert.h>
+#include <stdlib.h>
+
+typedef struct Node_T Node_T;
+
+struct Node_T
+{
+  int val;
+  struct Node_T *next;
+};
+
+struct Test
+{
+  Node_T *lists[4];
+};
+
+void test(struct Test Test)
+{
+  assert(Test.lists[1]->next);
```

Comment:
We could also check for the other indices 0, 2, and 3, and enable `--bounds-check` and then also access `Test.lists[4]` to check that an out of bounds failure is still reported. In addition we could also set `--max-nondet-tree-depth 2` and then add assertions `assert(Test.lists[<n>]->next->next == NULL)` to check that the array members are initialized with `gen_nondet_init()` just like ordinary variables.

Reply:
Very much in favour of this.

Reply:
I already did this